File: Day204_2025/Q204_GFG.java
class Q204_GFG {
    public int subarraySum(int[] arr) {
        int n = arr.length;
        int sum = 0;

        for (int i = 0; i < n; i++) {
            int start = i + 1;
            int end = n - i;
            int contribution = start * end;
            sum += arr[i] * contribution;
        }

        return sum;
    }
}

File: Day204_2025/Q204_GFG.cpp
class Solution {
  public:
    int subarraySum(vector<int>& arr) {
        // Q. Sum of Subarrays
        // code here
        int n=arr.size();
        int sum=0;
        
        for(int i=0; i<n; i++){
            int start=i+1;
            int end=n-i;
            int contri=start*end;
            sum+=(arr[i]*contri);
        }
        
        return sum;
    }
};